presto-spi/src/main/java/com/facebook/presto/spi/type/AbstractType.java

```diff
@@ -84,6 +84,9 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
     @Override
     public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
     {
+        if (!isOrderable()) {
+            throw new UnsupportedOperationException(getTypeSignature() + " type is not ordered");
+        }
```

Comment:
Remove these 3 lines, and add

    if (!elementType.isOrderable()) {
        throw new UnsupportedOperationException(getTypeSignature() + " type is not ordered");
    }

to the beginning of `ArrayType.compareTo`.